modules/ting_das/includes/ting_das.client.inc

```diff
@@ -0,0 +1,430 @@
+<?php
+
+/**
+ * @file
+ * Client to communicate with the Digital Article Services (DAS).
+ */
+
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\Client;
+
+/**
+ * Class TingDasCopyRequest.
+ */
+class TingDasCopyRequest {
+
+  private $url;
+  private $wsUsername;
+  private $wsPassword;
+
+  private $pid = NULL;
+  private $userLoanerId = NULL;
+  private $userName = NULL;
+  private $mail = NULL;
+  private $userInterestDays = NULL;
+  private $pickupAgencyId = NULL;
+  private $agencyId = NULL;
+  private $publicationTitle = NULL;
+  private $publicationDateOfComponent = NULL;
+  private $publicationYearOfComponent = NULL;
+  private $issueOfComponent = NULL;
+  private $volumeOfComponent = NULL;
+  private $pagesOfComponent = NULL;
+
+  /**
+   * TingDasCopyRequest constructor.
+   *
+   * @param string $url
+   *   The service endpoint for digital article service.
+   * @param string $username
+   *   The username to login with.
+   * @param string $password
+   *   The password to login with.
+   */
+  public function __construct(string $url, string $username, string $password) {
+    $this->url = $url;
+    $this->wsUsername = $username;
+    $this->wsPassword = $password;
+  }
+
+  /**
+   * Set date well id.
+   *
+   * @param string $pid
+   *   The date well id for the material requested.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setPid(string $pid) {
+    $this->pid = $pid;
+    return $this;
+  }
+
+  /**
+   * The data well id.
+   *
+   * @return string|null
+   *   The data well pid if set else NULL.
+   */
+  public function getPid() {
```

Comment:
All getters in this class are only called internally where you might as well use the properties. If that is the case then you might as well drop the accessors as they have little value.

Reply:
As the service is un-documented all the getters are here for completeness nothing else. The scope and usage for the service in the feature is unknown.

I you insist I'll remove them?